src/hw/raspi4/uart.c

```diff
@@ -0,0 +1,76 @@
+#include "port/raspi4/gpio.h"
+#include "port/raspi4/mbox.h"
+
+/* PL011 UART registers */
+#define UART0_DR   ((volatile unsigned int*)(MMIO_BASE + 0x00201000))
+#define UART0_FR   ((volatile unsigned int*)(MMIO_BASE + 0x00201018))
+#define UART0_IBRD ((volatile unsigned int*)(MMIO_BASE + 0x00201024))
+#define UART0_FBRD ((volatile unsigned int*)(MMIO_BASE + 0x00201028))
+#define UART0_LCRH ((volatile unsigned int*)(MMIO_BASE + 0x0020102C))
+#define UART0_CR   ((volatile unsigned int*)(MMIO_BASE + 0x00201030))
+#define UART0_IMSC ((volatile unsigned int*)(MMIO_BASE + 0x00201038))
+#define UART0_ICR  ((volatile unsigned int*)(MMIO_BASE + 0x00201044))
+
+void uart_init() {
+  register unsigned int r;
```

Comment:
`register` seems unneeded.

If it's needed to point codegen in the right direction for some reason, I'd prefer explicit use of inline asm. Or at least a good comment why it's needed.